docs/meshstack.meshmarketplace.metrics-metering.md

```diff
@@ -0,0 +1,441 @@
+- Hard-disk space used by a service instance: The service broker will report the used hard disk space with a timestamp when this value was observed. In that case it is recommended to use a not too frequent sampling of the metric. A new value would only be provided every hour or every day, as e.g. new values every second would be too fine-grained and result in too much data traffic and processing. If you need this kind of precision for a use-case, better provide a [Periodic Counter](#periodic-counters). That way you can already summarize the metric for a certain period.
+
+The cost for a gauge **must be an HOURLY cost**, for example the hourly cost of a small virtual machine.
+
+For implementation details of a gauge metrics endpoint, please have a look at this [section](#providing-gauges).
+
+### Periodic Counters
+
+A PeriodicCounter can be used to model counts which belong to time periods. For example, between `2020-09-01T00:00:00.000Z` and `2020-09-12T00:00:00.000Z` there were 10,000 API requests.
+
+**Applicable use-cases:**
+
+- The service instance shall be billed with an invoice from a third party. A service broker uses another service in the background like Datadog or MongoDB Atlas. These services will provide invoices by themself and these shall be charged to the consumer of the service.
+- Number of API calls done in an API gateway service (assuming the service keeps count in time periods)
+- incoming/outgoing traffic in gigabytes (assuming the service keeps count in time periods)
+- any other scenario where a counter is needed, and counter resets are a possibility. That means it is not an ever increasing counter, but e.g. it is reset monthly. If it is an ever increasing counter, use [Sampling Counters](#sampling-counters) instead.
+
+The cost for a PeriodCounter metric must be the cost per count. For example the cost per API call (or if the service counts API call in thousands, the cost per thousand API calls).
+
+For implementation details of a periodic counter metrics endpoint, please have a look at this [section](#providing-periodic-counters).
+
+### Sampling Counters
+
+A sampling counter can be used to model a counter that reports a count along with a timestamp at which the count was observed. For example, at `2020-09-11T00:00:00.000Z` it was observed that the amount of outgoing traffic was at 105GB.  **The value of a sampling counter should monotonically increase over time. Any counter resets are not handled in meshStack. If you expect counter resets you can use a [Periodic Counter](#periodic-counters) instead.**
+
+In general you can always track sampling counter metrics also as periodic counters. But sampling counters might be easier to implement on service broker side. You don't have to keep track of periods by yourself. This is handled by meshStack for you.
+
+> Sampling Counters are already collected by meshStack, but not yet processed to Tenant Usage Reports. This will be supported soon.
```

Comment:
What I miss with these 'coming soon' messages is if it is recommended to already do the implementation or not. I get the sense that I can do it already but I'm not sure if e.g. the data will correctly end up in the TURPs in the end.

Reply:
added it